Packs/FireEyeEX/Integrations/FireEyeEX/FireEyeEX.py

```diff
@@ -0,0 +1,575 @@
+    exist = False
+    current_allowed_list = client.fe_client.list_blockedlist_request(type_)
+    for entry in current_allowed_list:
+        if entry_value == entry.get('name'):
+            exist = True
+    if not exist:
+        raise DemistoException(str(f'Cannot delete the entry_value {entry_value} as it does not exist in the '
+                                   f'Blockedlist of type {type_}.'))
+
+    # gets 200 back without content if successful
+    client.fe_client.delete_blockedlist_request(type_, entry_value)
+
+    return CommandResults(
+        readable_output=f'Blockedlist entry {entry_value} of type {type_} was deleted.'
+    )
+
+
+@logger
+def fetch_incidents(client: Client, last_run: dict, first_fetch: str, max_fetch: int = 50,
+                    info_level: str = 'concise') -> Tuple[dict, list]:
+    if not last_run:  # if first time fetching
+        next_run = {
+            'time': to_fe_datetime_converter(first_fetch),
+            'last_alert_ids': []
+        }
+    else:
+        next_run = last_run
+    demisto.info(f'{INTEGRATION_NAME} executing fetch with: {str(next_run.get("time"))}')
+
+    raw_response = client.fe_client.get_alerts_request(request_params={
+        'start_time': to_fe_datetime_converter(first_fetch),
+        'info_level': info_level
+    })
+    all_alerts = raw_response.get('alert')
+
+    if not all_alerts:
+        demisto.info(f'{INTEGRATION_NAME} no alerts were fetched at: {str(next_run)}')
+        # as no alerts occurred till now, update last_run time accordingly
+        last_run['time'] = to_fe_datetime_converter('now')
+        return last_run, []
+
+    alerts = all_alerts[:max_fetch]
+    last_alert_ids = last_run.get('last_alert_ids', [])
+    if last_alert_ids:
+        last_alert_ids = json.loads(last_alert_ids)  # unescape the dumped json
+    incidents = []
+
+    for alert in alerts:
+        alert_id = str(alert.get('id'))
+        if alert_id not in last_alert_ids:  # check that event was not fetched in the last fetch
+            incident = {
+                'name': f'{INTEGRATION_NAME} Alert: {alert_id}',
+                'occurred': dateparser.parse(alert.get('occurred')).strftime(DATE_FORMAT),
+                'severity': alert_severity_to_dbot_score(alert.get('severity')),
+                'rawJSON': json.dumps(alert)
+            }
+            incidents.append(incident)
+            last_alert_ids.append(alert_id)
+
+    if not incidents:
+        demisto.info(f'{INTEGRATION_NAME} no new alerts were fetched at: {str(next_run)}')
+        # as no alerts occurred till now, update last_run time accordingly
+        last_run['time'] = to_fe_datetime_converter('now')
+        return last_run, []
+
+    # as alerts occurred till now, update last_run time accordingly to the that of latest fetched alert
+    next_run = {
+        'time': alerts[-1].get('occurred'),
+        'last_alert_ids': json.dumps(last_alert_ids)  # save the alert IDs from the last fetch
```

Comment:
in wat case `last_alert_ids` is being cleaned?

Reply:
currently it does not, just as we did in [Zimperium](https://github.com/demisto/content/blob/master/Packs/Zimperium/Integrations/Zimperium/Zimperium.py#L610) and in arcsight as well IIRC.